app/experimenter/experiments/models/nimbus.py

```diff
@@ -113,6 +113,9 @@ def __str__(self):
     def get_absolute_url(self):
         return reverse("nimbus-detail", kwargs={"slug": self.slug})
 
+    def has_state(self, state):
+        return type(self).objects.filter(id=self.id).filter(state).exists()
```

Comment:
This seems kind of roundabout, though: We have the experiment and its data fetched from the DB, but we're doing another DB query to determine whether it has a particular state. Maybe it'll make more sense if/when we use these for aggregate queries? Though that seems covered in the `{launch,pause,end}_queue` methods

Reply:
Yeah, not a colossal change either way, just kind of projecting forward. I wouldn't (haven't) r- on it so we can burn that bridge when we get to it.